pkg/reconciler/reconciler.go

```diff
@@ -591,6 +604,31 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Re
 		}
 	}()
 
+	if r.pauseReconcileAnnotation != "" {
+		if v, ok := obj.GetAnnotations()[r.pauseReconcileAnnotation]; ok {
+			if v == "true" {
+				log.Info(fmt.Sprintf("Resource has '%s' annotation set to 'true', reconcile paused.", r.pauseReconcileAnnotation))
+				u.UpdateStatus(
+					updater.EnsureCondition(conditions.Paused(corev1.ConditionTrue, conditions.ReasonPauseReconcileAnnotationTrue, "")),
+					updater.EnsureConditionUnknown(conditions.TypeIrreconcilable),
+					updater.EnsureConditionUnknown(conditions.TypeDeployed),
+					updater.EnsureConditionUnknown(conditions.TypeInitialized),
+					updater.EnsureConditionUnknown(conditions.TypeReleaseFailed),
+					updater.EnsureDeployedRelease(nil),
+				)
+				return ctrl.Result{}, nil
+			}
+		}
+	}
+
+	u.UpdateStatus(
+		// TODO(ROX-12637): change to updater.EnsureCondition(conditions.Paused(corev1.ConditionFalse, "", "")))
+		// once stackrox operator with pause support is released.
+		// At that time also add `Paused` to the list of conditions expected in stackrox operator e2e tests.
+		// Otherwise, the number of conditions in the `status.conditions` list will vary depending on the version
+		// of used operator, which is cumbersome due to https://github.com/kudobuilder/kuttl/issues/76
```

Comment:
@porridge is this comment still relevant?
I suppose we should remove references to stackrox in this PR...

Reply:
Unfortunately the comment is still relevant.

See also https://issues.redhat.com/browse/ROX-27939 Although the solution mentioned there might not be too practical, because of  https://github.com/kudobuilder/kuttl/issues/504. I think something like https://github.com/kudobuilder/kuttl/pull/388 would be best.

One might think it would be fine to just add this new condition to our tests, but this will not be easy due to upgrade tests which need to assert against both new and old operator code.

However I don't think any of this should stop upstreaming this change. Let's change this line as suggested in the TODO, and for the time being we'll change it back to `EnsureConditionAbsent` downstream. A one-line patch is still better than having to carry this whole change.